src/Lib/Canvas.vala

```diff
@@ -493,6 +496,20 @@ public class Akira.Lib.Canvas : Goo.Canvas {
         }
     }
 
+    /*
+     * Will update snap decorators if necessary.
+     */
+    private void on_update_snap_decorators () {
+        var extra_context = mode_manager.active_mode_extra_context ();
+        if (extra_context is Akira.Lib.Modes.TransformMode.TransformExtraContext) {
+            var tec = extra_context as Akira.Lib.Modes.TransformMode.TransformExtraContext;
+            snap_manager.generate_decorators (tec.snap_guide_data);
```

Comment:
You could type cast this directly when generating the decorators:
`snap_manager.generate_decorators (((Lib.Modes.TransformMode.TransformExtraContext) extra_context).snap_guide_data);`

Also, in general, you can avoid using the `Akira.` namespace and make the code a little shorter.

Reply:
I separate it more for readability (still made the change)